turbopuffer/namespace.py

```diff
@@ -89,7 +89,8 @@ def approx_count(self) -> int:
     def upsert(self,
                ids: Union[List[int], List[str]],
                vectors: List[List[float]],
-               attributes: Optional[Dict[str, List[Optional[str]]]] = None) -> None:
+               attributes: Optional[Dict[str, List[Optional[str]]]] = None,
+               distance_metric: Optional[str] = 'cosine_distance') -> None:
```

Comment:
probably reasonable to assume that people want cosine_distance, might be confusing if you don't set this at upsert and then later try to query with a different distance metric without realizing you're stuck with cosine

Reply:
ooops, that was an accident! good catch thx